go-controller/pkg/util/pod_annotation.go

```diff
@@ -380,7 +357,32 @@ func SecondaryNetworkPodIPs(pod *v1.Pod, networkInfo NetInfo) ([]net.IP, error)
 	return ips, nil
 }
 
+func GetPrimaryNetworkNADNameForPodFromNetInfo(pod *v1.Pod, netinfo NetInfo) (string, error) {
+	podNetworks, err := UnmarshalPodAnnotationAllNetworks(pod.Annotations)
+	if err != nil {
+		return "", err
+	}
+	for _, nadName := range netinfo.GetNADs() {
```

Comment:
So `nadName` is a namespaced name: `<namespace>/<name>`. 

So you can just parse the namespace and match it against the pod namespace.

I don't think we need to check the role. If we have the pod annotated with a non-primary NAD that we have as primary in NetInfo, we have bigger problems.

Reply:
I have asked @trozet to add a `GetNADs(namespace string)` method to avoid having to parse the namespaced name however.